core/src/test/java/org/apache/calcite/rex/RexBuilderTest.java

```diff
@@ -957,6 +957,16 @@ private void checkDate(RexLiteral literal) {
     assertThat(inCall.getKind(), is(SqlKind.SEARCH));
   }
 
+  @Test void testMakeInSingleValue() {
```

Comment:
Add jira link

Reply:
The JIRA link doesn't add much info. It is clear why the test is added from the git history.

Reply:
Git history can get lost if the code is significantly altered.

The point of this test is to check that if `makeIn` has a single argument the result is not a sarg, and that is not explicit.

It's a seven line test but which is the key line that demonstrates the bug? Is it that we used `SqlTypeName.INTEGER` rather than `SqlTypeName.SMALLINT`? As test author you know that six of those lines are boilerplate, and that before the bug (actually feature request) was fixed the string would have been "SEARCH($0, Sarg([100, 100]))" and that string is still in some sense correct but "=($0, 100)" is in some sense better.

We don't necessarily need a jira link but we do need more on the page (probably javadoc, or possibly a message in the `assertThat`). 

I am a huge fan of using javadoc to state the almost obvious - what is obvious to the author but not obvious to the reader.

Reply:
I tried to make the purpose of the test more explicit in https://github.com/apache/calcite/pull/3991/commits/d6ade5afaf85254e14ff1e60121a87c8e6feb224. Please check.